zerver/lib/bugdown/fenced_code.py

```diff
@@ -252,6 +264,37 @@ def done(self) -> None:
         self.output.append('')
         self.processor.pop()
 
+
+class ZpoilerHandler(BaseHandler):
+    def __init__(self, processor: Any, output: MutableSequence[str],
+                 fence: str, zpoiler_header: str) -> None:
+        self.processor = processor
+        self.output = output
+        self.fence = fence
+        self.zpoiler_header = zpoiler_header
+        self.lines: List[str] = []
+
+    def handle_line(self, line: str) -> None:
+        if line.rstrip() == self.fence:
+            self.done()
+        else:
+            check_for_new_fence(self.processor, self.lines, line)
+
+    def done(self) -> None:
+        if len(self.lines) == 0:
+            # No content, do nothing
+            return
+        else:
+            header = self.zpoiler_header if self.zpoiler_header != '' else "Spoiler"
```

Comment:
"Spoiler" should ideally probably be tagged for i18n.  Probably the nicest way to do this is for this logic to live client-side, so that we translate "Spoiler" in the client-side UI, rather than just using the language of whoever posted the spoiler message.